tests/prague/eip7702_set_code_tx/test_set_code_txs.py

```diff
@@ -3367,3 +3367,93 @@ def test_many_delegations(
         tx=tx,
         post=post,
     )
+
+
+def test_invalid_transaction_after_authorization(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+):
+    """
+    Test an invalid block due to a transaction reusing the same nonce as an authorization
+    included in a prior transaction.
+    """
+    auth_signer = pre.fund_eoa()
+
+    txs = [
+        Transaction(
+            sender=pre.fund_eoa(),
+            gas_limit=500_000,
+            to=Address(0),
+            value=0,
+            authorization_list=[
+                AuthorizationTuple(
+                    address=Address(1),
+                    nonce=0,
+                    signer=auth_signer,
+                ),
+            ],
+        ),
+        Transaction(
+            sender=auth_signer,  # Nonce is not bumped by the authorization, so it is still 0
```

Comment:
Since you are testing nonce here, I would suggest explicitly specify its value.

And for auto nonce calculation, we can have  a unit test. (Actually needed)

Reply:
Added in latest commit.